classes/Context.php

```diff
@@ -348,32 +348,35 @@ public function getTranslator()
         }
 
         $cacheDir = _PS_CACHE_DIR_.'translations';
-        $this->translator = new Translator($this->language->locale, null, $cacheDir, false);
+        $translator = new Translator($this->language->locale, null, $cacheDir, false);
 
         // In case we have at least 1 translated message, we return the current translator.
         // If some translations are missing, clear cache
-        if (count($this->translator->getCatalogue($this->language->locale)->all())) {
-            return $this->translator;
+        if (count($translator->getCatalogue($this->language->locale)->all())) {
+            $this->translator = $translator;
+            return $translator;
         }
 
         // However, in some case, even empty catalog were stored in the cache and then used as-is.
         // For this one, we drop the cache and try to regenerate it.
-        $cache_file = Finder::create()
-            ->files()
-            ->in($cacheDir)
-            ->depth('==0')
-            ->name('*.'.$this->language->locale.'.*');
-        (new Filesystem())->remove($cache_file);
+        if (is_dir($cacheDir)) {
```

Comment:
Is there a case where this folder does not exist?

Reply:
When you are clearing your cache.